components/org.wso2.carbon.identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java

```diff
@@ -62,36 +65,30 @@
 
-        tenantDomain =  ((AuthenticationContext) context.getHttpServletRequest().getAttribute("auth-context"))
-                .getUser().getTenantDomain();
-        ;
-        resourceOwnerName = ((AuthenticationContext) context.getHttpServletRequest().getAttribute("auth-context"))
-                .getUser().getUserName();
-        consumerKey = (String) ((AuthenticationContext) context.getHttpServletRequest().getAttribute("auth-context"))
-                .getParameter("consumer-key");
-        Response response = null;
-
-        if (requestedResource == null) {
-            log.error("Request body cannot be empty.");
-            return Response.status(Response.Status.BAD_REQUEST).build();
-        }
-        try {
-            Resource registerResource = ResourceUtils.getResourceService()
-                    .registerResource(ResourceUtils.getResource(requestedResource), resourceOwnerName, tenantDomain
-                            , consumerKey);
-            CreateResourceDTO createResourceDTO = ResourceUtils.createResponse(registerResource);
-            return Response.status(Response.Status.CREATED).entity(createResourceDTO).build();
-
-        } catch (UMAServiceException e) {
-            handleErrorResponse(e, log);
-        } catch (UMAClientException e) {
-            handleErrorResponse(e, log);
-            log.error("Client error when retrieving resource from the resource server.", e);
-        } catch (Throwable throwable) {
-            handleErrorResponse((UMAException) throwable, log);
-            log.error("Internal server error occurred. ", throwable);
+        if (isValidToken(context)) {
+            if (requestedResource.getResourceScopes() == null ||
+                    CollectionUtils.isEmpty(requestedResource.getResourceScopes())) {
+                log.error("Request body cannot be empty and should consist with scopes.");
+                return Response.status(Response.Status.BAD_REQUEST).build();
+            }
+            try {
+                Resource registerResource = ResourceUtils.getResourceService()
+                        .registerResource(ResourceUtils.getResource(requestedResource),
+                                getResourceOwner(context), PrivilegedCarbonContext.getThreadLocalCarbonContext()
+                                        .getTenantId(), getConsumerKey(context));
+                CreateResourceDTO createResourceDTO = ResourceUtils.createResponse(registerResource);
+                return Response.status(Response.Status.CREATED).entity(createResourceDTO).build();
+
+            } catch (UMAServiceException e) {
+                handleErrorResponse(e, true);
+            } catch (UMAClientException e) {
+                log.error("Client error when retrieving resource from the resource server.", e);
+                handleErrorResponse(e, false);
+            }
         }
-        return response;
+        log.error("Resource can not be retrieved due to client side error.");
```

Comment:
No need to have a error log. And the log message seems to be incorrect. This is executed if isValidToken is failed.